### ISSUE TITLE
Bug: Starting a new application takes user back to existing application

### ISSUE THREAD

**Issue by igorkorenfeld** [cisagov/getgov#394]
When I try to start a 2nd new application the button on the dashboard takes me into an existing application.

To replicate
1. Click "Start a new domain request" on the dashboard
2. Fill out the first few pages of the form to make it distinct 
3. Navigate back to home 
4. Click "Start a new domain request" on the dashboard
5. Check the first few page you filled out before to confirm this is the same application

Additionally, other bugs associated with this request:

AC:

- Update functionality to allow for multiple applications in the same session

See: https://gsa-tts.slack.com/archives/C03QM0JGSQG/p1675441631322649

A note for anyone picking this up to fix it: this is more challenging than it first appears. The current application is being saved in state to allow a user to navigate between the pages of the application without losing progress. We don't have any ability to know when they have "exited" the application. (We should avoid any JavaScript hacks to try to "guess" since all of these approaches are buggy and rely on the user having JavaScript enabled/working.) We could *very easily* reset state each time they visit the "front page" of an application (as they do when clicking "apply"), however, we would then need to ensure they had some way of getting back to previously saved applications (since the "front page" will now always show them a fresh blank application).

The code to fix this bug might be pretty simple, but there are plenty of ramifications to think through. 🙂

**Comment by abroddrick** [comment]
Designer considerations @PaulKuykendall @Katherine-Osos

**Comment by gabydisarli** [comment]
My immediate thoughts are your second suggestion Alysia — to warn the user that they have one in progress and giving the option to either resume or start a new application. This has its own implications to consider. Design will review this and discuss next week.

**Comment by Katherine-Osos** [comment]
As Gaby noted, the designers will discuss on Monday. But in preparation, I'm curious about a couple things: why there isn't a "delete" option for applications that have not been submitted yet. Are there technical reasons why that can't be included? Applicants, could in theory, have a couple apps started and are waiting to confirm a few details before submitting. So "resume current" could apply to multiple apps (and there's already an "Edit" button next to each in progress app that does the same thing.

**Comment by rachidatecs** [comment]
- There's currently no technical barrier preventing a user from submitting multiple requests.
- Coming in fresh and clicking START NEW APPLICATION will start a new application. Clicking the button again in the same session will edit the application that was already started. Clicking START NEW APPLICATION in a new session (I used another browser) will indeed start a new application.
- Clicking on an EDIT button will reset the application tracker (the locks in the sidenav) but will always pull the correct data from the DB

To recap: 

- The app keeps track of one in-progress application in any given session, but can pull fresh data from the DB. Clicking EDIT, you'll start on screen one with the right data. 
- Clicking START NEW APPLICATION in a fresh session will start a new application, no matter how many other active applications you may have.
- Clicking START NEW APPLICATION in a session where you have already started an application OR edited an existing one, you'll just load the latest application you were looking at.

**Comment by rachidatecs** [comment]
The 3 viable solutions I see are either tracking all applications in session, tracking all applications in the DB (persistent), or creating 2 start pages for the form, one for new applications and one for existing applications. The first 2 are probably expensive and I'm not sure about the viability and scope of the last one. I recommend we discuss this week, with the 18F engineering team present.

**Comment by rachidatecs** [comment]
One in-progress application will require putting some new guardrails on the user model and the Manage your domains UI (and some UI/UX revisions there as well). We could/should probably leave the setpper form and session storage alone.

Special Analyst tooling: can't speak to that without more details.

**Comment by gabydisarli** [comment]
There's another level of nuance here that I think we're still missing and I didn't touch on in my earlier comment. From what I'm seeing, there are two user groups:

User group 1: new registrants who have never submitted an application. These users should only be allowed one in-progress application. 
User group 2: registrants who have submitted an application and had it approved. These users can submit more than one application. 

I'm getting this from our status transitions doc, specifically:


> ### **Domain applications**
> **Started**
> An applicant has begun to fill in their data to request a .gov domain.
> 
> Actions available:
> 
> - Edit any fields of the application (subject to normal user permissions)
> - Delete application
> 
> 
> Actions forbidden:
> 
> - Edit nameservers and DNSSEC data for unapproved domain
> - Add security email for unapproved domain
> - Add domain management contacts for unapproved domain
> 
> [none] → started
> Preconditions:
> 
> - User is logged in
> - User has either 0 applications or 1+ approved applications _(user with no previously approved applications cannot submit more than one application)_

**Comment by gabydisarli** [comment]
Noting that this bug is 🚫blocked🚫 until we have further discussion with business, product, and design to decide on user group experiences

**Comment by h-m-f-t** [comment]
Thanks for the bump, Gabe. Your [comment last week](https://github.com/cisagov/getgov/issues/394#issuecomment-1589511288) gets this exactly right.
